modules/tensor_mechanics/src/materials/czm/SalehaniIrani3DCTraction.C

```diff
@@ -0,0 +1,159 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "SalehaniIrani3DCTraction.h"
+
+registerMooseObject("TensorMechanicsApp", SalehaniIrani3DCTraction);
+
+template <>
+InputParameters
+validParams<SalehaniIrani3DCTraction>()
+{
+  InputParameters params = validParams<CZMMaterialBase>();
+  params.addClassDescription("3D Coupled (3DC) cohesive law of Salehani and Irani with no damage");
+  params.addRequiredParam<Real>(
+      "normal_gap_at_maximum_normal_traction",
+      "the value of normal gap at which maximum normal traction is achieved");
+  params.addRequiredParam<Real>(
+      "tangential_gap_at_maximum_shear_traction",
+      "the value of tangential gap at which maximum shear traction is achieved");
+  params.addRequiredParam<Real>("maximum_normal_traction",
+                                "The maximum normal traction the interface can sustain");
+  params.addRequiredParam<Real>("maximum_shear_traction",
+                                "The maximum shear traction the interface can sustain");
+  return params;
+}
+
+SalehaniIrani3DCTraction::SalehaniIrani3DCTraction(const InputParameters & parameters)
+  : CZMMaterialBase(parameters), _deltaU0(3, 0), _maxAllowableTraction(3, 0)
+{
+  const_cast<std::vector<Real> &>(_deltaU0)[0] =
+      getParam<Real>("normal_gap_at_maximum_normal_traction");
+  const_cast<std::vector<Real> &>(_maxAllowableTraction)[0] =
+      getParam<Real>("maximum_normal_traction");
+
+  for (unsigned int i = 1; i < 3; i++)
+  {
+    const_cast<std::vector<Real> &>(_deltaU0)[i] =
+        getParam<Real>("tangential_gap_at_maximum_shear_traction") * std::sqrt(2);
+    const_cast<std::vector<Real> &>(_maxAllowableTraction)[i] =
+        getParam<Real>("maximum_shear_traction");
+  }
+}
+
+RealVectorValue
+SalehaniIrani3DCTraction::computeTraction()
+{
+  // traction convention N, T, S, where N is the noraml direction and T and S
+  // two arbitrarty tangential driection
+  RealVectorValue traction_local;
+
+  // just a temporaty containers for auxiliary calcualtions
+  Real aa, X, expX, A_i, B_i;
+  // indecx variables to avaoid multiple redefinition
```

Comment:
Spell check

Reply:
Follow MOOSE variable naming conventions, and don't capitalize these:
X->x, expX->exp_x, A_i->a_i, B_i->b_i